Adding issue template to bug reports in Monero repository.

0xFFFC0000 opened the pull request:
Includes two templates: Bug Report, and Feature Request.

In case you want to look at how it appears in a GitHub report page, hit the `New issue` button here [1]. In addition to the template, this will assign `bug` label to bug reports and `feature` label to feature requests.


1. https://github.com/0xFFFC0000/monero-perf/issues

selsta commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,93 @@
+      description: |
+        Please copy and paste any relevant log output or attach a debug log file.
+
+        You can get log information via --log-level=4 flag or by taking a look at log files at ~/.bitmonero/ in cli version. Or you can look at the log information in Settings->Log section in Monero-GUI.
```
I don't think we should require log-level 4 here, it's extremely noisy, 2 is better for most issues.

0xFFFC0000 replied:
Consider it done. Will change this to level 2.

0xFFFC0000 replied:
Fixed it in a new push. We have log-level=2 now.

Gingeropolous commented:
I would include a link to search for existing issues when you ask the user if they've checked. 

honestly, I've never liked these templates. I understand their efficiency, but there have been many cases where I'm using open source software, and I come across an issue, so I head to the github to file an issue, and I'm hit with a way-too-involved issue template so I just don't file it. ("I'm an end user, not a debugger!")

and some of these fields will get repetitive for _someone_ who regularly uses monero and finds issues.  :)

but this seems less onerous than some templates I've found out there. At least only some of the fields are required.

0xFFFC0000 commented:
@Gingeropolous I agree.  It is understandable for users to decide to not report issues for projects that do have many extra ***mandatory*** templates. But the key here is the **mandatory** word. Without extra, useless mandatory fields, it is quite useful. Without forcing us to put in extra time, these templates do provide much better support. For example, we have users reporting a compilation bug, but they don't even mention what is the os they are using. 


Will add a link to search existing issues.

selsta commented:
Personally I would prefer the following fields, it's a good tradeoff between having necessary information and complexity of creating the issue

- describe the issue
- logs
- which version
- where did you get the binary
- which OS, hardware

selsta commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,93 @@
+
+        * General monero questions and/or support requests should use Monero StackExchange at https://monero.stackexchange.com.
+        * For reporting security issues, please read instructions at https://github.com/monero-project/meta/blob/master/VULNERABILITY_RESPONSE_PROCESS.md.
+
```
Please also add a point that GUI related issues should be reported on the GUI repo.

0xFFFC0000 replied:
Will be added in new PR. Thanks.

selsta commented on .github/ISSUE_TEMPLATE/bug_report.yml:
```diff
@@ -0,0 +1,93 @@
+      description: |
+        Please copy and paste any relevant log output or attach a debug log file.
+
+        You can get log information via --log-level=2 flag or by taking a look at log files at ~/.bitmonero/ in cli version. Or you can look at the log information in Settings->Log section in Monero-GUI.
```
Should we leave this in? For example the wallet log file is in a different place than the daemon log file, also --log-level=2 affects the log that gets printed and written to the disk, so that "or" doesn't really make sense.

0xFFFC0000 replied:
I added a more detailed comment here in new push. Please don't hesitate to let me know if it needs to change again. Thanks

0xFFFC0000 commented:
Changelog for last push: 

1. Changed `Expected behaviour` to a non-mandatory field.  
2. Changed `Machine specifications` to mandatory field.
3. Added point about the Monero-GUI repository issue tracker.
4. Improved the comment about how can user generate comment.



Example: https://github.com/0xFFFC0000/monero-perf/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml

0xFFFC0000 committed:
Adding issue templates to bug reports in Monero repository.
Includes two templates: Bug Report, and Feature Request.
